### PR TITLE
Reduce GitHub Token Permissions

### PR DESCRIPTION
**Pull request by k-doering-NOAA**
I recently learned that it is possible to be specific about the [permissions for GITHUB_TOKEN](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token) and that most workflows don't require that many permissions. Reducing permissions follows the rule of least privilege and reduces the security risk of using GitHub actions (e.g., if a third party action were hacked).

I went through and curated permissions for GitHub tokens in the ghactions4r reusable workflows to the minimum required. I tested all to make sure that they will still run (on the "permissions" branch, and in a separate repository).

Users should not experience any change in functionality from this Pull Request.

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
@Bai-Li-NOAA if you don't mind taking a quick look to make sure I didn't make an obvious mistake, that would be great! Then, I think this can be squashed and merged.

File: .github/workflows/update-pkgdown.yml [hunk 1]
```diff
@@ -18,6 +18,24 @@ on:
 
 name: update-pkgdown
 
+# Give the fewest permissions possible. content is necessary.
+# https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token
+permissions:
+  contents: write
+  pull-requests: none
```
Comment (Bai-Li-NOAA) [inline .github/workflows/update-pkgdown.yml]: Non-blocking suggestion/question: If you specify access for contents, do you need to explicitly set none for other permissions, or are unspecified permissions automatically set to none?

File: .github/workflows/doc-and-style-r.yml [hunk 2]
```diff
@@ -27,6 +27,24 @@ on:
        type: boolean
        default: false
 
+# Give the fewest permissions possible. content and pull-requests are necessary.
+# https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token
+permissions:
+  contents: write
+  pull-requests: write
+  actions: none
```
Comment (Bai-Li-NOAA) [inline .github/workflows/doc-and-style-r.yml]: Non-blocking suggestion/question: If you specify access for contents and pull-requests, do you need to explicitly set none for other permissions, or are unspecified permissions automatically set to none?

**Review by Bai-Li-NOAA — APPROVED** [review]
I don't know anything about permissions, so I'm glad to learn something new today! The PR looks good to me, I just have some minor suggestions/questions!

File: .github/workflows/update-pkgdown.yml (hunk 1 above)
Comment (k-doering-NOAA) [reply .github/workflows/update-pkgdown.yml]: Yes, I think you have to specifically set them to none for others- I think contents isn't everything, I think it is just the code part of the repository.

Comment (k-doering-NOAA) [reply .github/workflows/update-pkgdown.yml]: actually - I think you are right - the docs say " If you specify the access for any of these permissions, all of those that are not specified are set to none." Thanks for pointing this out! Still, I think it is more clear to leave them all for now, if that is ok!

By the way, if you don't specify any of this, these are the defaults: https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Comment (k-doering-NOAA) [reply .github/workflows/update-pkgdown.yml]: On second thought, I will simplify the permissions! I also did a test to confirm you don't need all the "none" lines. Thank you so much for the suggestion!

**Comment by k-doering-NOAA** [comment]
@Bai-Li-NOAA thank you so much for the suggestion! As usual, I think a PR "didn't really need to be reviewed", but of course it did - thank you for the helpful feedback and quick review! I will merge once the checks pass.
